File: src/main/java/pl/training/workshop/m6/s12_onemany/start/CancellationDesk.java
package pl.training.workshop.m6.s12_onemany.start;

import java.time.LocalDateTime;
import java.util.List;

import pl.training.workshop.m6.s12_onemany.TicketData;
import pl.training.workshop.shared.Money;

/** Start: klient - kasa zwrotów. Sam rozróżnia jeden bilet i wiele biletów. */
public final class CancellationDesk {
    private final RefundService service = new RefundService();

    public Money refund(List<TicketData> tickets, LocalDateTime now) {
        if (tickets.size() == 1) {
            return service.refund(tickets.getFirst(), now);
        }
        return service.refundAll(tickets, now);
    }
}

File: typescript/src/workshop/m6/s12_onemany/start/CancellationDesk.ts
import type { Money } from '../../../shared/Money.js';
import type { LocalDateTime } from '../../../shared/time.js';
import type { TicketData } from '../TicketData.js';
import { RefundService } from './RefundService.js';

/** Start: klient - kasa zwrotów. Sam rozróżnia jeden bilet i wiele biletów. */
export class CancellationDesk {
  private readonly service = new RefundService();

  refund(tickets: readonly TicketData[], now: LocalDateTime): Money {
    const [first] = tickets;
    if (tickets.length === 1 && first !== undefined) {
      return this.service.refund(first, now);
    }
    return this.service.refundAll(tickets, now);
  }
}
